src/map_field.cpp

```diff
@@ -154,97 +171,188 @@ static int ter_furn_movecost( const ter_t &ter, const furn_t &furn )
-                            scents.set_unsafe( p, std::max( assignment[x][y].intensity, scents.get_unsafe( p ) ) );
-                            break;
-                        }
-                    }
+    }
+
+    auto neighs = get_neighbors( p );
+    size_t end_it = static_cast<size_t>( rng( 0, neighs.size() - 1 ) );
+    std::vector<size_t> spread;
+    std::vector<maptile> neighbour_vec;
+    // Then, spread to a nearby point.
+    // If not possible (or randomly), try to spread up
+    // Wind direction will block the field spreading into the wind.
+    spread.reserve( 8 );
+    // Start at end_it + 1, then wrap around until all elements have been processed.
+    for( size_t i = ( end_it + 1 ) % neighs.size(), count = 0 ;
+         count != neighs.size();
+         i = ( i + 1 ) % neighs.size(), count++ ) {
+        const auto &neigh = neighs[i];
+        if( gas_can_spread_to( cur, neigh ) ) {
+            spread.push_back( i );
+        }
+    }
+    auto maptiles = get_wind_blockers( winddirection, p );
+    maptile remove_tile = std::get<0>( maptiles );
+    maptile remove_tile2 = std::get<1>( maptiles );
+    maptile remove_tile3 = std::get<2>( maptiles );
+    // three map tiles that are facing th wind direction.
+    if( !zlevels || one_in( spread.size() ) ) {
+        // Construct the destination from offset and p
+        if( g->is_sheltered( p ) || windpower < 5 ) {
+            gas_spread_to( cur, neighs[ random_entry( spread ) ] );
+        } else {
+            end_it = static_cast<size_t>( rng( 0, neighs.size() - 1 ) );
+            // Start at end_it + 1, then wrap around until all elements have been processed.
+            for( size_t i = ( end_it + 1 ) % neighs.size(), count = 0 ;
+                 count != neighs.size();
+                 i = ( i + 1 ) % neighs.size(), count++ ) {
+                const auto &neigh = neighs[i];
+                if( ( neigh.x != remove_tile.x && neigh.y != remove_tile.y ) ||
+                    ( neigh.x != remove_tile2.x && neigh.y != remove_tile2.y ) ||
+                    ( neigh.x != remove_tile3.x && neigh.y != remove_tile3.y ) ) {
+                    neighbour_vec.push_back( neigh );
+                } else if( x_in_y( 1, std::max( 2, windpower ) ) ) {
+                    neighbour_vec.push_back( neigh );
                 }
             }
+            if( !neighbour_vec.empty() ) {
+                gas_spread_to( cur, neighbour_vec[rng( 0, neighbour_vec.size() - 1 )] );
+            }
+        }
+    } else if( zlevels && p.z < OVERMAP_HEIGHT ) {
+        tripoint up{p.x, p.y, p.z + 1};
+        maptile up_tile = maptile_at_internal( up );
+        if( gas_can_spread_to( cur, up_tile ) && valid_move( p, up, true, true ) ) {
+            gas_spread_to( cur, up_tile );
         }
     }
+};
 
-    point index( const tripoint &p ) const {
-        return point( p.x - origin.x, p.y - origin.y );
+/*
+Function: create_hot_air
```

Comment:
Why are you repeating the name of the function in the comment?

Reply:
I've just moved this code block, but yeah comments should be revised.